shared-libs/couch-request/src/couch-request.js

```diff
@@ -0,0 +1,84 @@
+const request = require('request-promise-native');
+
+const isPlainObject = (value) => {
+  if (typeof value !== 'object' || value == null) {
+    return false;
+  }
+  const prototype = Object.getPrototypeOf(value);
+  return (prototype === null || prototype === Object.prototype || Object.getPrototypeOf(prototype) === null)
+    && !(Symbol.toStringTag in value) && !(Symbol.iterator in value);
+};
+
+const memoGlobalCouchRequestOptions = () => {
+  const cache = new Map(); // the assumption here is that this will be a fairly hot path, so caching would be valuable
+  return (COUCH_URL) => {
+    if (cache.has(COUCH_URL)) {
```

Comment:
The first time through I thought this was checking the wrong key name but it works because you set both. I think this would be more readable.

```suggestion
    if (cache.has('hostname')) {
```

Reply:
This is using the value as the key to avoid the value comparison check, so it has to be the same as the passed value from `process.env.COUCH_URL`

I can change the variable name to lower case. I think I was trying to imply that it was an env variable but I can see how that is confusing. Would that be adequate?

Reply:
I don't understand why you need to set `COUCH_URL` AND `hostname` as `hostname` is the only one that's read. If you want to avoid checking again then check if `hostname` is set.

Reply:
I see what you're saying.